### PR TITLE
I've revised the original example for calling `queryEvents` with the Sui TypeScript SDK.

### PR DESCRIPTION
**Pull request by baicaiyihao**
## Description 

I've revised the original example for calling `queryEvents` with the Sui TypeScript SDK.

## Test Plan 

I believe that the previous example of using queryEvents in the Sui TypeScript SDK is outdated, and the import statement import { JsonRpcProvider, testnetConnection } from '@mysten/sui.js'; is no longer valid. This can mislead beginners and make it difficult for them to learn. As a beginner myself, I was only able to successfully retrieve the events from a transaction by examining multiple layers of the source code through the client. Therefore, I hope the example can be updated so that beginners can simply and correctly use queryEvents to query the contents of events in a transaction.

https://docs.sui.io/guides/developer/sui-101/using-events

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/guides/developer/sui-101/using-events.mdx [hunk 1]
```diff
@@ -127,64 +127,95 @@ This example leverages the Sui TypeScript SDK to subscribe to events the package
...
+        </Flex>
+        </>
+    );
+}
 ```
 
 ### Response
 
-When the subscribed to event fires, the example displays the following JSON representation of the event.
+Example transaction：https://suiexplorer.com/txblock/2hg7QQvNh7DnmmMaanAZ7hRYCAUimPJTN3bJRnpo29ej?network=testnet
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/using-events.mdx:216]:
```suggestion
When the subscribed to event fires, the example displays the following JSON representation of the event.

{
  "id": {
    "txDigest": "2hg7QQvNh7DnmmMaanAZ7hRYCAUimPJTN3bJRnpo29ej",
    "eventSeq": "0"
  },
  "packageId": "<PACKAGE-ID>",
  "transactionModule": "<MODULE>",
  "sender": "<SENDER-ADDRESS>",
  "type": "<PACKAGE-ID>::<MODULE>::Result",
  "parsedJson": {
    "msg": "You Win :)"
  },
  "bcs": "3ZrJhsA3GhA8gKr",
  "timestampMs": "1707120095029"
}
```
Can't use a link to testnet as data is not guaranteed to persist.

File: docs/content/guides/developer/sui-101/using-events.mdx [hunk 2]
```diff
@@ -127,64 +127,95 @@ This example leverages the Sui TypeScript SDK to subscribe to events the package
...
-	}
-});
+```ts
+import {
+    useSignAndExecuteTransactionBlock,
+    useSuiClient,
+  } from "@mysten/dapp-kit";
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { Button, Flex, Heading } from "@radix-ui/themes";
+import { SICBOGAME_PACKAGE_ID } from "./constants";
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/using-events.mdx:141]:
```suggestion

```

File: docs/content/guides/developer/sui-101/using-events.mdx [hunk 3]
```diff
@@ -127,64 +127,95 @@ This example leverages the Sui TypeScript SDK to subscribe to events the package
...
+    const { mutate: signAndExecute } = useSignAndExecuteTransactionBlock();
+    const client = useSuiClient();
+
+    const executeMoveCall = (method: "small" | "large") => {
+        const txb = new TransactionBlock();
+
+        if (method === "small") {
+        txb.moveCall({
+            arguments: [txb.pure.u64(0)],
+            target: `${SICBOGAME_PACKAGE_ID}::sicbogame::create_game`,
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/using-events.mdx:157]:
```suggestion
            target: `<PACKAGE-ID>::<MODULE>::create_game`,
```

File: docs/content/guides/developer/sui-101/using-events.mdx [hunk 4]
```diff
@@ -127,64 +127,95 @@ This example leverages the Sui TypeScript SDK to subscribe to events the package
...
+        });
+        } else {
+        txb.moveCall({
+            arguments: [txb.pure.u64(1)],
+            target: `${SICBOGAME_PACKAGE_ID}::sicbogame::create_game`,
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/using-events.mdx:162]:
```suggestion
            target: `<PACKAGE-ID>::<MODULE>::create_game`,
```

File: docs/content/guides/developer/sui-101/using-events.mdx [hunk 5]
```diff
@@ -127,64 +127,95 @@ This example leverages the Sui TypeScript SDK to subscribe to events the package
...
-    "bet_amount": "4000000000",
-    "game_id": "0xa7e1fb3c18a88d048b75532de219645410705fa48bfb8b13e8dbdbb7f4b9bbce",
-    "guess": 0,
-    "player_won": true
-  },
-  "bcs": "3oWWjWKRVu115bnnZphyDcJ8EyF9X4pgVguwhEtcsVpBf74B6RywQupm2X",
-  "timestampMs": "1687912116638"
-}
-```
+I can get the value of parsedJson in the example below, and its final value is {"msg":"You Win :)"} .
```
Comment (ronny-mysten) [inline docs/content/guides/developer/sui-101/using-events.mdx:218]:
```suggestion
As the returned JSON shows, the value of `parsedJson` is `{"msg":"You Win :)"}` .
```

**Review by ronny-mysten — CHANGES_REQUESTED** [review]
Thanks @baicaiyihao ! @damir still needs to review but I did have a few necessary changes.

File: docs/content/guides/developer/sui-101/using-events.mdx (hunk 5 above)
Comment (baicaiyihao) [reply docs/content/guides/developer/sui-101/using-events.mdx:218]: Thanks @ronny-mysten ，you're probably right.

**Pull request closed without merging**